SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -259,17 +260,75 @@ void ASpatialDebugger::OnEntityRemoved(const Worker_EntityId EntityId)
+
+	if (bAuthoritative && AuthOp.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
+	{
+		if (NetDriver->VirtualWorkerTranslator == nullptr)
+		{
+			// Currently, there's nothing to display in the debugger other than load balancing information.
+			return;
+		}
+
+		VirtualWorkerId LocalVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+		FColor LocalVirtualWorkerColor = SpatialGDK::GetColorForWorkerName(NetDriver->VirtualWorkerTranslator->GetLocalPhysicalWorkerName());
+
+		SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(AuthOp.entity_id);
+		if (DebuggingInfo == nullptr)
+		{
+			// Some entities won't have debug info, so create it now.
+			SpatialDebugging NewDebuggingInfo(LocalVirtualWorkerId, LocalVirtualWorkerColor, SpatialConstants::INVALID_VIRTUAL_WORKER_ID, InvalidServerTintColor, false);
+			Worker_ComponentData DebuggingData = NewDebuggingInfo.CreateSpatialDebuggingData();
+			NetDriver->Connection->SendAddComponent(AuthOp.entity_id, &DebuggingData);
+			return;
+		}
+		else
+		{
+			DebuggingInfo->AuthoritativeVirtualWorkerId = LocalVirtualWorkerId;
+			DebuggingInfo->AuthoritativeColor = LocalVirtualWorkerColor;
+			Worker_ComponentUpdate DebuggingUpdate = DebuggingInfo->CreateSpatialDebuggingUpdate();
+			NetDriver->Connection->SendComponentUpdate(AuthOp.entity_id, &DebuggingUpdate);
+		}
+	}
+}
+
+void ASpatialDebugger::ActorAuthorityIntentChanged(Worker_EntityId EntityId, VirtualWorkerId NewIntentVirtualWorkerId) const
+{
+	SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(EntityId);
+	check(DebuggingInfo != nullptr);
+	DebuggingInfo->IntentVirtualWorkerId = NewIntentVirtualWorkerId;
+
+	const PhysicalWorkerName* NewAuthoritativePhysicalWorkerName = NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(NewIntentVirtualWorkerId);
+	check(NewAuthoritativePhysicalWorkerName != nullptr);
+
+	DebuggingInfo->IntentColor = SpatialGDK::GetColorForWorkerName(*NewAuthoritativePhysicalWorkerName);
+	Worker_ComponentUpdate DebuggingUpdate = DebuggingInfo->CreateSpatialDebuggingUpdate();
+	NetDriver->Connection->SendComponentUpdate(EntityId, &DebuggingUpdate);
+
+}
+
+
 void ASpatialDebugger::DrawTag(UCanvas* Canvas, const FVector2D& ScreenLocation, const Worker_EntityId EntityId, const FString& ActorName)
 {
 	SCOPE_CYCLE_COUNTER(STAT_DrawTag);
 
 	// TODO: Smarter positioning of elements so they're centered no matter how many are enabled https://improbableio.atlassian.net/browse/UNR-2360.
 	int32 HorizontalOffset = -32.0f;
 
+	check(NetDriver != nullptr && !NetDriver->IsServer());
+	if (!NetDriver->StaticComponentView->HasComponent(EntityId, SpatialConstants::SPATIAL_DEBUGGING_COMPONENT_ID))
+	{
+		return;
+	}
+
+	const SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(EntityId);
```

Comment:
I personally prefer this `return;` early layout, however, I think GDK style would be to put the assignment and everything following in an if statement (see https://docs.improbable.io/unreal/alpha/contributions/unreal-gdk-coding-standards), i.e. 
```
if (const SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(EntityId)) 
{
      	if (bShowLock)
        ...
}

```

Reply:
I think that's mostly about setting the value of things that are created outside of a conditional, but in this case the assignment is just for querying.